src/modules/Exchange.ts

```diff
@@ -508,9 +500,13 @@ export class Exchange {
     },
     options?: CallOptions,
   ): Promise<string> {
-    const token = new this.contracts.web3.eth.Contract((erc20Abi as Json).abi, tokenAddress);
+    const token = this.contracts.collateralToken;
+    token.options.address = tokenAddress;
```

Comment:
I can't tell what this change was needed for. The new version is bad since you're overriding `this.contracts.collateralToken.address` so if you try to use `this.contracts.collateralToken` in the future it won't actually be the collateral token